Dockerfile

```diff
@@ -1,17 +1,34 @@
-FROM node:12-alpine
-
+FROM node:12-alpine as base
 WORKDIR /app
-
 COPY package*.json ./
+ENV PORT=8080
 
-RUN npm install
-
+FROM base AS dev
+RUN npm ci
 COPY . .
+CMD ["npm", "run", "dev"]
 
+FROM dev AS test
+COPY . .
+CMD ["npm", "run", "test"]
```

Comment:
This may look like major overkill here, but the purpose is for ci/cd environments where you can spin up a separate lint and unit/integration test image without making the dev or prod images depend on it.

Reply:
Actually better practice might be to use RUN instead of CMD so it blocks building of the image if checks dont pass. ¯\_(ツ)_/¯